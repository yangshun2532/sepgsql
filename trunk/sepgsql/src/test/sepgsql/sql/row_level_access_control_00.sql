-- 
-- testcases for row-level access controls (initial setup)
-- 
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0-s0:c0.c15

CREATE TABLE t1
(
	x	integer primary key,
	y	text
);

CREATE TABLE t2
(
	a	integer references t1(x),	
	b	text
);

CREATE TABLE t3
(
	s	integer,
	t	text
);

INSERT INTO t1 VALUES	(1, 'water'), (2, 'coke'), (3, 'juice'), (4, 'coffee'),
       	       		(5, 'sake'), (6, 'beer'), (7, 'wine'), (8, 'red tea');
INSERT INTO t2 VALUES	(2, 'aaa'), (3, 'bbb'), (4, 'ccc');
INSERT INTO t3 VALUES	(10, 'red'), (11, 'green'), (12, 'blue'), (13, 'orange'),
       	       		(14, 'yellow'), (15, 'white'), (16, 'black');

UPDATE t1 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE x IN (2, 4, 6, 8);
UPDATE t2 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE a IN (2, 3);
UPDATE t3 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE s IN (10, 13, 14);
UPDATE t3 SET security_context = sepgsql_set_range(security_context, 's0:c1')
       WHERE s IN (15, 16);

SELECT security_context, * FROM t1;

SELECT security_context, * FROM t2;

SELECT security_context, * FROM t3;
