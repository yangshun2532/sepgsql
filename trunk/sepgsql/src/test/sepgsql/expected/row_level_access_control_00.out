-- 
-- testcases for row-level access controls (initial setup)
-- 
-- selinux_user: unconfined_u
-- selinux_role: unconfined_r
-- selinux_type: unconfined_t
-- selinux_range: s0-s0:c0.c15
CREATE TABLE t1
(
	x	integer primary key,
	y	text
);
psql:./row_level_access_control_00.sql:13: NOTICE:  CREATE TABLE / PRIMARY KEY will create implicit index "t1_pkey" for table "t1"
CREATE TABLE t2
(
	a	integer references t1(x),	
	b	text
);
CREATE TABLE t3
(
	s	integer,
	t	text
);
INSERT INTO t1 VALUES	(1, 'water'), (2, 'coke'), (3, 'juice'), (4, 'coffee'),
       	       		(5, 'sake'), (6, 'beer'), (7, 'wine'), (8, 'red tea');
INSERT INTO t2 VALUES	(2, 'aaa'), (3, 'bbb'), (4, 'ccc');
INSERT INTO t3 VALUES	(10, 'red'), (11, 'green'), (12, 'blue'), (13, 'orange'),
       	       		(14, 'yellow'), (15, 'white'), (16, 'black');
UPDATE t1 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE x IN (2, 4, 6, 8);
UPDATE t2 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE a IN (2, 3);
UPDATE t3 SET security_context = sepgsql_set_range(security_context, 's0:c0')
       WHERE s IN (10, 13, 14);
UPDATE t3 SET security_context = sepgsql_set_range(security_context, 's0:c1')
       WHERE s IN (15, 16);
SELECT security_context, * FROM t1;
              security_context               | x |    y    
---------------------------------------------+---+---------
 unconfined_u:object_r:sepgsql_table_t:s0    | 1 | water
 unconfined_u:object_r:sepgsql_table_t:s0    | 3 | juice
 unconfined_u:object_r:sepgsql_table_t:s0    | 5 | sake
 unconfined_u:object_r:sepgsql_table_t:s0    | 7 | wine
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 2 | coke
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 4 | coffee
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 6 | beer
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 8 | red tea
(8 rows)

SELECT security_context, * FROM t2;
              security_context               | a |  b  
---------------------------------------------+---+-----
 unconfined_u:object_r:sepgsql_table_t:s0    | 4 | ccc
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 2 | aaa
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 3 | bbb
(3 rows)

SELECT security_context, * FROM t3;
              security_context               | s  |   t    
---------------------------------------------+----+--------
 unconfined_u:object_r:sepgsql_table_t:s0    | 11 | green
 unconfined_u:object_r:sepgsql_table_t:s0    | 12 | blue
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 10 | red
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 13 | orange
 unconfined_u:object_r:sepgsql_table_t:s0:c0 | 14 | yellow
 unconfined_u:object_r:sepgsql_table_t:s0:c1 | 15 | white
 unconfined_u:object_r:sepgsql_table_t:s0:c1 | 16 | black
(7 rows)

